Configuration/Skimming/src/LeptonSkimming.cc

```diff
@@ -0,0 +1,487 @@
+   for (const reco::Muon& mu: *muons){
+      double tempDR=deltaR(mu.eta(),mu.phi(),trk.eta(),trk.phi());
+      if (minDR<tempDR) continue;
+      minDR=tempDR;
+   }
+   if (minDR<MuTrkMinDR_Cut) continue;
+   if (SelectedMu_DR<1000 ){
+     if (fabs(ZvertexTrg-trk.vz())>TrackMuDz_Cut ) continue;
+     if ( deltaR(trk.eta(),trk.phi(),SelectedTrgObj_PtEtaPhiCharge[1],SelectedTrgObj_PtEtaPhiCharge[2])<TrgExclusionCone) continue;
+   }
+   //assignments   
+   auto cleanTrack=std::make_shared<reco::Track>(trk);
+   cleanedTracks.push_back(cleanTrack);
+   Trk_container.push_back(trk_index);
+   trk_index++;
+ }
+ 
+    //create mother ee combination
+ 
+ // fit track pairs
+    cleanedObjTracks.clear(); cleanedPairTracks.clear();     
+    TLorentzVector vel1,vel2;
+    std::vector<std::shared_ptr<reco::Track>> cleanedObjects; 
+    //objects    
+    for(auto & vec: MuTracks) cleanedObjects.push_back(vec);
+    for(auto & vec: ElTracks) cleanedObjects.push_back(vec);  
+
+    if (cleanedObjects.size()==0)
+             return false;
+
+    for(auto& obj: cleanedObjects){
+      //      auto obj=cleanedObjects.at(iobj);
+      auto tranobj=std::make_shared<reco::TransientTrack>(reco::TransientTrack(*obj,&(*bFieldHandle)));
+      unsigned int iobj=&obj-&cleanedObjects[0];
+      unsigned int index=object_container.at(iobj);
+      if ( object_id.at(iobj)==13 && QualMu_Cut==1 && !muon_soft.at(index)) continue;
+     if ( object_id.at(iobj)==13 && QualMu_Cut==2 && !muon_medium.at(index)) continue;
+     if ( object_id.at(iobj)==13 && QualMu_Cut==3 && !muon_tight.at(index)) continue;
+     if (object_id.at(iobj)==13) vel1.SetPtEtaPhiM(muon_pt.at(index),muon_eta.at(index),muon_phi.at(index),0.0005);
+       
+     else vel1.SetPtEtaPhiM(el_pt.at(index),el_eta.at(index),el_phi.at(index),0.0005);
+    
+     
+      if (object_id.at(iobj)==13 && vel1.Pt()<PtMu_Cut) continue;
+      for(auto& trk2: cleanedTracks){
+ 	unsigned int itrk2=&trk2-&cleanedTracks[0];
+	//auto trk2=cleanedTracks.at(itrk2);
+         if (obj->charge()*trk2->charge()==1) continue;
+         if (ObjPtLargerThanTrack && vel1.Pt()<trk2->pt()) continue;
+         vel2.SetPtEtaPhiM(trk2->pt(),trk2->eta(),trk2->phi(),0.0005);
+         if (deltaR(obj->eta(),obj->phi(),trk2->eta(),trk2->phi())<TrkObjExclusionCone) continue;
+         if (object_id.at(iobj)==13 && deltaR(obj->eta(),obj->phi(),SelectedTrgObj_PtEtaPhiCharge[1],SelectedTrgObj_PtEtaPhiCharge[2])<MuTrgExclusionCone) continue;
+         if (object_id.at(iobj)==11 && deltaR(obj->eta(),obj->phi(),SelectedTrgObj_PtEtaPhiCharge[1],SelectedTrgObj_PtEtaPhiCharge[2])<ElTrgExclusionCone) continue;
+         if (SelectedMu_DR<1000 ){
+	   if (object_id.at(iobj)==13 && fabs(ZvertexTrg- obj->vz())>MuTrgMuDz_Cut ) continue;
+           if (object_id.at(iobj)==11 && fabs(ZvertexTrg-obj->vz())>ElTrgMuDz_Cut ) continue;
+         }
+         if ((vel1+vel2).M()>MaxMee_Cut || (vel1+vel2).M()<MinMee_Cut ) continue;        
```

Comment:
better to pre-compute m^2 and use it instead of sqrt(m^2)

Reply:
I wanted to ask something on that. When you say pre-compute you mean to create external function that gives m^2? If so is this somewhere implemented or i have to write it?

Reply:
@gkaratha I assume that @davidlange6 means computing just once .M2()
(see https://root.cern.ch/doc/master/TLorentzVector_8h_source.html#l00491 ) and using that local value for the comparison